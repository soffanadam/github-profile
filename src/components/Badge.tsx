import React from 'react'

export type BadgeProps = {
  dataTestID?: string
}

export const Badge: React.FC<BadgeProps> = ({ dataTestID, children }) => {
  return (
    <span
      data-testid={dataTestID}
      className="inline-flex items-center py-0.5 px-2.5 text-gray-500 bg-white rounded border"
    >
      {children}
    </span>
  )
}
