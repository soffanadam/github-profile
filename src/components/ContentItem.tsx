import { Content, ContentType } from '@/types'
import React from 'react'
import { FcFile, FcFolder, FcLink, FcPackage } from 'react-icons/fc'

export const ICON_TEST_ID = 'icon'

export type ContentItemProps = {
  dataTestID?: string
  content: Content
}

export const ContentItem: React.FC<ContentItemProps> = ({
  dataTestID,
  content
}) => {
  return (
    <span
      data-testid={dataTestID}
      className="flex items-center py-2 px-5 hover:bg-gray-50 border-b"
    >
      <span className="w-8">
        {content.type == ContentType.DIR ? (
          <FcFolder data-icon={ContentType.DIR} />
        ) : content.type == ContentType.SYMLINK ? (
          <FcLink data-icon={ContentType.SYMLINK} />
        ) : content.type == ContentType.SUBMODULE ? (
          <FcPackage data-icon={ContentType.SUBMODULE} />
        ) : (
          <FcFile data-icon={ContentType.FILE} />
        )}
      </span>
      {content.name}
    </span>
  )
}
