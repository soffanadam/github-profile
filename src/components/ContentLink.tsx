import { Content, ContentType, PlainObject } from '@/types'
import React from 'react'
import { Link } from 'react-router-dom'

export type ContentLinkProps = {
  dataTestID?: string
  repoUrl: string
  content: Content
}

export const ContentLink: React.FC<ContentLinkProps & PlainObject> = ({
  dataTestID,
  repoUrl,
  content,
  children,
  ...rest
}) => {
  if (content.type === ContentType.DIR) {
    return (
      <Link
        data-testid={dataTestID}
        to={`${repoUrl}/${content.path}`}
        {...rest}
      >
        {children}
      </Link>
    )
  }

  if (content.html_url) {
    return (
      <a
        data-testid={dataTestID}
        href={content.html_url}
        target="_blank"
        rel="noreferrer"
        {...rest}
      >
        {children}
      </a>
    )
  }

  return (
    <div data-testid={dataTestID} {...rest}>
      {children}
    </div>
  )
}
