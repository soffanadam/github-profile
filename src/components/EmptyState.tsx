import { classNames } from '@/utils'
import React from 'react'

export type EmptyStateProps = {
  dataTestID?: string
  className?: string
}

export const EmptyState: React.FC<EmptyStateProps> = ({
  children,
  dataTestID,
  className
}) => {
  return (
    <div
      data-testid={dataTestID}
      className={classNames('py-3 px-5 text-xl', className)}
    >
      {children}
    </div>
  )
}
