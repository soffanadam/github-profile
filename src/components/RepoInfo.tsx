import { Repo } from '@/types'
import React from 'react'
import { Link } from 'react-router-dom'

export type RepoInfoProps = {
  dataTestID?: string
  repo: Repo
}

export const RepoInfo: React.FC<RepoInfoProps> = ({ dataTestID, repo }) => {
  return (
    <Link
      data-testid={dataTestID}
      to={`/${repo.owner.login}/${repo.name}`}
      className="group block"
    >
      <span
        data-testid="repo-info-name"
        className="inline-flex pb-1 mb-2 text-2xl text-indigo-500 border-b border-transparent group-hover:border-indigo-300"
      >
        {repo.name}
      </span>
      {repo.description && (
        <p data-testid="repo-info-description" className="mb-5">
          {repo.description}
        </p>
      )}
    </Link>
  )
}
