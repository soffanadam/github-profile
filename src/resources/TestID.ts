export enum TestID {
  SEARCH_FORM = 'search-form',
  SEARCH_FORM_INPUT = 'search-form-input'
}
