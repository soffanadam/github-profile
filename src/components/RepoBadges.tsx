import { Repo } from '@/types'
import React from 'react'
import { FiStar, FiGitBranch } from 'react-icons/fi'
import { Badge } from './Badge'

export type RepoBadgesProps = {
  repo: Repo
}

export const RepoBadges: React.FC<RepoBadgesProps> = ({ repo }) => {
  return (
    <div className="flex mt-2 space-x-3">
      <Badge>
        <FiStar className="mr-2" />
        <span data-stargazers-count>{repo.stargazers_count}</span>
      </Badge>
      <Badge>
        <FiGitBranch className="mr-2" />
        <span data-forks-count>{repo.forks_count}</span>
      </Badge>
    </div>
  )
}
