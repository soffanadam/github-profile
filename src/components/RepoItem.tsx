import { Repo } from '@/types'
import React from 'react'
import { FiStar } from 'react-icons/fi'
import { FiGitBranch } from 'react-icons/fi'
import { Link } from 'react-router-dom'
import { Badge } from './Badge'

export type UserRouteProps = {
  repo: Repo
}

export const RepoItem: React.FC<UserRouteProps> = ({ repo }) => {
  return (
    <div className="py-3 px-5 rounded shadow hover:shadow-md transition transform hover:scale-105">
      <Link
        to={`/${repo.owner.login}/${repo.name}`}
        className="block text-2xl text-indigo-500"
      >
        {repo.name}
      </Link>
      {repo.description && <p>{repo.description}</p>}
      <div className="flex mt-5 space-x-3">
        <Badge>
          <FiStar className="mr-2" />
          {repo.stargazers_count}
        </Badge>
        <Badge>
          <FiGitBranch className="mr-2" />
          {repo.forks_count}
        </Badge>
      </div>
    </div>
  )
}
