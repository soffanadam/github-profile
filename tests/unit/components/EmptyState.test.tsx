import React from 'react'
import { render } from '@testing-library/react'
import { EmptyState, EmptyStateProps } from '@/components/EmptyState'
import { TestID } from '@/resources/TestID'
import { withEmptyStateProps } from './utils'

// constants
const TEXT = 'text'

const renderComponent = (props: EmptyStateProps) => {
  return render(<EmptyState {...props}>{TEXT}</EmptyState>)
}

describe('<EmptyState />', () => {
  it('renders correctly', () => {
    const props = withEmptyStateProps()
    const component = renderComponent(props)
    expect(component).toBeTruthy()

    const testElement = component.getByTestId(TestID.EMPTY_STATE)
    expect(testElement).toHaveTextContent(TEXT)
    expect(testElement).toHaveClass(props.className)
    expect(testElement).toMatchSnapshot()
  })
})
