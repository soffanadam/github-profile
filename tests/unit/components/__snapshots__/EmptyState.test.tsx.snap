// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState /> renders correctly 1`] = `
<div
  class="py-3 px-5 text-xl mt-4"
  data-testid="empty-state"
>
  text
</div>
`;
